### PR TITLE
Using Visual Studio BuildTools as a MinGW alternative

### PR DESCRIPTION
**Pull request by l0rd**
Building the [MSI hook on Windows](`contrib/win-installer/podman-msihooks/check.c) currently requires [MinGW CC](https://www.mingw-w64.org/). This commit updates the build script (build-hooks.bat is now build-hooks.ps1) so that, when MinGW is absent but the C compiler included in Visual Studio BuildTools is installed, the latter is used to build the MSI hook.

Documentation in `build_windows.md` has been updated with the instructions to install VS BuildTools (not MSYS2/MinGW).

Another change included in the PR is the `installertest` target in windows Makefile (`winmake.ps1`) to run the Windows installer tests that are part of Cirrus CI.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: contrib/win-installer/test-installer.ps1
Comment (lsm5) [inline contrib/win-installer/test-installer.ps1:27]: @l0rd these 3 lines seem to be failing tests.

Comment (l0rd) [reply contrib/win-installer/test-installer.ps1:27]: @lsm5 thank you. I suspect that's related to the version of PowerShell that's used by CI (< v7.0) that doesn't support the [ternary operator syntax](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_if?view=powershell-7.4#using-the-ternary-operator-syntax) I am going to put this PR back in draft mode until I fix it.

Comment (lsm5) [reply contrib/win-installer/test-installer.ps1:27]: are you planning to update the cirrus windows image with latest powershell? Looks like github actions windows env does have [required min-version of powershell](https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#powershell-tools)

Comment (l0rd) [reply contrib/win-installer/test-installer.ps1:27]: This is an option, yes. But no, I think it's simpler to update the PowerShell script introduced with this PR to support old and new versions of PWSH. So that it has a wider compatibility of windows dev environments.

**Comment by l0rd** [comment]
Tests are passing. Re-opening for review. @containers/podman-maintainers PTAL.

**Comment by mheon** [comment]
LGTM, though my unfamiliarity with Powershell makes me less able to find issues than normal

**Review by lsm5 — APPROVED** [review]
I removed mingw tools from my windows env and installed the VSSetup extension as mentioned in the PR. Was able to build the installer just fine.

I won't be able to review the powershell details, but I can say it works for me :) . Setting a hold in case anyone else has the time/expertise to review it.

@n1hility PTAL

/lgtm
/approve
/hold

**Pull request merged**
